src/main/java/seedu/address/model/listing/UniqueListingList.java

```diff
@@ -0,0 +1,137 @@
+package seedu.address.model.listing;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.listing.exceptions.DuplicateListingException;
+import seedu.address.model.listing.exceptions.ListingNotFoundException;
+
+/**
+ * A list of listing that enforces uniqueness between its elements and does not allow nulls.
+ * A listing is considered unique by comparing using {@code Listing#isSameListing(Listing)}. As such, adding and
+ * updating of persons uses Listing#isSameListing(Listing) for equality so as to ensure that the listing being added or
+ * updated is unique in terms of identity in the UniqueListingList. However, the removal of a listing uses
+ * Listing#equals(Object) so as to ensure that the listing with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Listing#isSameListing(Listing)
+ */
+public class UniqueListingList implements Iterable<Listing> {
+
+    private final ObservableList<Listing> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Listing> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent person as the given argument.
+     */
+    public boolean contains(Listing toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameListing);
+    }
+
+    /**
+     * Adds a person to the list.
+     * The person must not already exist in the list.
+     */
+    public void add(Listing toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateListingException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
```

Comment:
there are a few more persons in here and other javaDocs that supposed to be listing